### PR TITLE
Replace current_peset_num with 'n' which should be the active pelist.

### LINKED ISSUE #1243: Wrong peset used in mpp_broadcast_
I suspect that the current_peset_num used in line 185 of mpp/include/mpp_transmit_mpi.fh should really be 'n'

I don't have a clean reproducer that demonstrates the issue.
I am testing the use of mpp_broadcast for communicating between parent and nested grids in the UFS/HAFS model.
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/272

I'm using a pelist in the mpp_broadcast() call.  The code in the above PR fails with "FATAL from PE    36: MPP_BROADCAST: broadcasting from invalid PE." unless I place mpp_set_current_pelist() calls judiciously around the mpp_broadcast() call.

While investigating the "broadcasting from invalid PE" message, I realized the error message is coming from a check to ensure "from_pe" is part of the "current_peset_num" NOT the peset that is retrieved via the call to get_peset().

**System Environment**
Describe the system environment, include:
- OS: Ubuntu and SLES15
- Compiler(s): Intel 2023.1 (aka 2021.9.0) and 19.1.3.304
- MPI type, and version OpenMPI-4.1.5 and Cray-MPICH
- netCDF Version: For both C and Fortran
- Configure options: Any additional flags, or macros passed to configure

I have run the FMS (2023.01) internal test suite via "make check" with and without the 'fix' below and both give the same results.

      !if( .NOT.ANY(from_pe.EQ.peset(current_peset_num)%list) ) &
      if( .NOT.ANY(from_pe.EQ.peset(n)%list) )

### PR DESCRIPTION
**Pull request by dkokron**
While chasing down the cause of the following error message, I realized that the check to ensure the transmitting rank is part of the peset is not using the peset returned by the get_peset call.

"FATAL from PE 36: MPP_BROADCAST: broadcasting from invalid PE."

I converted part of the UFS/HAFS application to use mpp_broadcast() instead of numerous point-to-point messages.  Testing with a HAFS case revealed the above error message.  I was able to make the code work by surrounding the mpp_broadcast() call with appropriate mpp_set_current_pelist() calls.  A colleague said these mpp_set_current_pelist() calls should not be needed and provided a unit test to prove it.

Further investigation without the mpp_set_current_pelist() scoping calls revealed the issue addressed with the PR.

Fixes #1243

Testing was performance under Ubuntu-22.04 using the Intel 2023.1 suite of compilers and OpenMPI-4.1.5.
I have run the FMS internal check (make check) with and without the change in this PR.  No change in the results.
I have also run "make distcheck".  The logs config, make, check and distcheck logs are attached.

[checkOPT.log](https://github.com/NOAA-GFDL/FMS/files/11659060/checkOPT.log)
[config.log](https://github.com/NOAA-GFDL/FMS/files/11659061/config.log)
[distcheckOPT.log](https://github.com/NOAA-GFDL/FMS/files/11659062/distcheckOPT.log)
[makeOPT.log](https://github.com/NOAA-GFDL/FMS/files/11659063/makeOPT.log)

I also ran a HAFS case with a single nest on the NOAA 'acorn' system.  This is the original case that started this whole investigation.  That case ran fine using an FMS with the change contained in his PR.

**Checklist:**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules
- [x] New check tests, if applicable, are included
- [x] `make distcheck` passes

### CODE REVIEW

**Comment by lharris4** [comment]
@laurenchilutti I know @bensonr is away for a couple of weeks. Is it possible someone else could review this change?

**Comment by laurenchilutti** [comment]
@lharris4 I will find someone in the group to review this.

**Review by rem1776 — DISMISSED** [review]

**Comment by rem1776** [comment]
> @lharris4 I will find someone in the group to review this.

@thomas-robinson will need to review this to merge since he's the other code owner for this file. I just approved as well, but doesn't make a difference merging-wise.

**Comment by lharris4** [comment]
Hi, all. Thanks.

**Comment by thomas-robinson** [comment]
We will discuss this today at our group meeting

**Comment by laurenchilutti** [comment]
@dkokron how immediately would you like to see this change released in FMS? Would mid July be too late?

**Comment by dkokron** [comment]
Hmm, that delays
https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/272 quite a bit.
I do have an alternative solution that uses MPI directly.  Hopefully the
model group will agree to use the MPI solution until FMS is available and
approved for use by NOAA.
Dan

**Review by thomas-robinson — DISMISSED** [review]

**Comment by thomas-robinson** [comment]
@dkokron We would prefer adding your single fix to a patch instead of using a workaround if the July timeline is too far in the future.  We should be able to sync it up with the merge of NOAA-GFDL/GFDL_atmos_cubed_sphere#272

**Comment by dkokron** [comment]
Based on feedback from the HAFS developers, I think it's best to wait for the next scheduled FMS release.

**Review by rem1776 — APPROVED** [review]

**Review by thomas-robinson — APPROVED** [review]

**Comment by thomas-robinson** [comment]
Thanks @dkokron we are going to put it into our next round of testing.

**Pull request merged**
